extensions/cache/runtime/src/main/java/io/quarkus/cache/runtime/devconsole/CacheDevConsoleRecorder.java

```diff
@@ -18,34 +22,34 @@
 public class CacheDevConsoleRecorder {
     public Handler<RoutingContext> clearCacheHandler() {
         return new DevConsolePostHandler() {
-            int code;
-            String message = "";
 
             @Override
-            protected void handlePost(RoutingContext event, MultiMap form) throws Exception {
+            protected void handlePost(RoutingContext event, MultiMap form) {
                 String cacheName = form.get("name");
                 Optional<Cache> cache = CaffeineCacheSupplier.cacheManager().getCache(cacheName);
                 if (cache.isPresent() && cache.get() instanceof CaffeineCache) {
                     CaffeineCacheImpl caffeineCache = (CaffeineCacheImpl) cache.get();
 
                     String action = form.get("action");
                     if (action.equalsIgnoreCase("clearCache")) {
-                        caffeineCache.invalidateAll();
+                        caffeineCache.invalidateAll().subscribe().with(ignored -> {
+                            endResponse(event, OK, createResponseMessage(caffeineCache));
+                        });
+                    } else if (action.equalsIgnoreCase("refresh")) {
+                        endResponse(event, OK, createResponseMessage(caffeineCache));
+                    } else {
+                        String errorMessage = "Invalid action: " + action;
+                        endResponse(event, INTERNAL_SERVER_ERROR, createResponseError(cacheName, errorMessage));
                     }
-                    this.code = HttpResponseStatus.OK.code();
-                    this.message = createResponseMessage(caffeineCache);
-                    return;
                 } else {
                     String errorMessage = "Cache for " + cacheName + " not found";
-                    this.code = HttpResponseStatus.NOT_FOUND.code();
-                    this.message = createResponseError(cacheName, errorMessage);
+                    endResponse(event, NOT_FOUND, createResponseError(cacheName, errorMessage));
                 }
             }
 
-            @Override
-            protected void actionSuccess(RoutingContext event) {
```

Comment:
I think that you should still override the `actionSuccess()` and keep the method body empty, otherwise the logic from the `DevConsolePostHandler.actionSuccess(RoutingContext)` can be used.

Reply:
I removed the empty method intentionally as the super method seemed harmless in this case because it starts with `if (!event.response().ended())`. Could you please confirm you want the empty method back anyway?

Reply:
I don't think it's harmless for the `clearCache` action - the response is not ended yet because you call `event.response().end()` in the callback, i.e. potentially after the `actionSuccess()` is invoked... hm, but vertx should throw an `IllegalStateException` if `end()` is called multiple times so I might be missing something as well ;-).

Reply:
Np :D

Reply:
PR updated.

Reply:
CC @cescoffier @jponge ^

Reply:
The callbacks of `with(...)` might be called just right after the subscription if data is already there or if the workload it triggers is synchronous (e.g. you compute some value in-memory and there's no async I/O call involved).

Reply:
Thanks @jponge.

In this case, there could be an async I/O call involved in the future (not right now) so keeping the empty method is definitely the best thing to do @mkouba.